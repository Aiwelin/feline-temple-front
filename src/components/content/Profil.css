#form-profil {
  width: 100%;

  .avatar {
    border-radius: 50%;
    float: left;
    padding-right: 2em;
  }
}
