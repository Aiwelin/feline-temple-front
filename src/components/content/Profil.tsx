import "./Profil.css";
import * as React from "react";
import { User } from "../../interface/UserInterface";
import { UserOutlined } from "@ant-design/icons";
import { Avatar, Form, Input } from "antd";

export default class Profil extends React.Component<User, {}> {
  constructor(props: User) {
    super(props);
  }
  render() {
    return (
      <div id="form-profil">
        <h2>Mon profil</h2>
        <Form layout="vertical" labelAlign="right">
          {this.props.avatar ? (
            <img key={this.props.avatar} src={this.props.avatar} />
          ) : (
            <Avatar size={64} icon={<UserOutlined />} />
          )}
          <Form.Item label="PrÃ©nom">
            <Input disabled value={this.props.name} />
          </Form.Item>
          <Form.Item label="Nom">
            <Input disabled value={this.props.lastName} />
          </Form.Item>
          <Form.Item label="Adresse">
            <Input disabled value={this.props.address} />
          </Form.Item>
        </Form>
      </div>
    );
  }
}
