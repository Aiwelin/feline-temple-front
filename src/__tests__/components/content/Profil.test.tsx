import { expect, it } from "vitest";
import Profil from "../../../components/content/Profil";

it("should work as expected with data", () => {
  const result = (
    <Profil
      address="18 RUE MAGDALINE, 44700 NANTES"
      avatar="src/assets/avatar/apichard.jpg"
      email="amandine.pichard.auditeur@lecnam.net"
      lastName="PICHARD"
      name="Amandine"
      password="Test@2023"
      isConnected
    />
  );
  expect(result).toMatchSnapshot();
});
