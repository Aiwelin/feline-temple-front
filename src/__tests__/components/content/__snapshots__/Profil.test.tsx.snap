// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should work as expected 1`] = `
<Profil
  address="18 RUE MAGDALINE, 44700 NANTES"
  avatar="src/assets/avatar/apichard.jpg"
  email="amandine.pichard.auditeur@lecnam.net"
  isConnected={true}
  lastName="PICHARD"
  name="Amandine"
  password="Test@2023"
/>
`;

exports[`should work as expected with data 1`] = `
<Profil
  address="18 RUE MAGDALINE, 44700 NANTES"
  avatar="src/assets/avatar/apichard.jpg"
  email="amandine.pichard.auditeur@lecnam.net"
  isConnected={true}
  lastName="PICHARD"
  name="Amandine"
  password="Test@2023"
/>
`;

exports[`should work as expected with no data 1`] = `<Profil />`;
